src/Microsoft.OData.Core/Json/ODataUtf8JsonWriter.Stream.cs

```diff
@@ -0,0 +1,347 @@
+            /// </summary>
+            public override long Position { get => throw new NotSupportedException(); set => throw new NotSupportedException(); }
+
+            /// <summary>
+            /// Flushes any buffered data to the underlying stream synchronously.
+            /// </summary>
+            public override void Flush()
+            {
+                this.jsonWriter.Flush();
+            }
+
+            /// <summary>
+            /// Flushes any buffered data to the underlying stream asynchronously.
+            /// </summary>
+            /// <param name="cancellationToken">A A cancellation token to observe while waiting for the flush operation to complete.</param>
+            /// <returns>A task representing the asynchronous flush operation.</returns>
+            public override async Task FlushAsync(CancellationToken cancellationToken)
+            {
+                await this.jsonWriter.FlushAsync().ConfigureAwait(false);
+            }
+
+            /// <summary>
+            /// Disposes the object.
+            /// </summary>
+            /// <param name="disposing">True if called from Dispose; false if called form the finalizer.</param>
+            protected override void Dispose(bool disposing)
+            {
+                if (this.numBytesNotWrittenFromPreviousChunk > 0)
+                {
+                    // If there are unprocessed bytes, encode and write them as the final block.
+                    ReadOnlySpan<byte> bytesNotProcessedFromPreviousChunk = this.buffer.AsSpan().Slice(0, this.numBytesNotWrittenFromPreviousChunk);
+
+                    this.jsonWriter.Base64EncodeAndWriteChunk(bytesNotProcessedFromPreviousChunk, isFinalBlock: true, out numBytesNotWrittenFromPreviousChunk);
+                    this.numBytesNotWrittenFromPreviousChunk = 0;
+                }
+
+                if (this.buffer != null)
+                {
+                    ArrayPool<byte>.Shared.Return(this.buffer);
+                }
+
+                this.Flush();
+                base.Dispose(disposing);
+            }
+
+            /// <summary>
+            /// Asynchronously disposes of the current object and performs cleanup operations.
+            /// </summary>
+            /// <returns>A <see cref="ValueTask"/> representing the asynchronous operation.</returns>
+            public override async ValueTask DisposeAsync()
+            {
+                if (this.numBytesNotWrittenFromPreviousChunk > 0)
+                {
+                    // If there are unprocessed bytes, encode and write them as the final block.
+                    ReadOnlyMemory<byte> bytesNotProcessedFromPreviousChunk = this.buffer.AsMemory().Slice(0, this.numBytesNotWrittenFromPreviousChunk);
+
+                    this.jsonWriter.Base64EncodeAndWriteChunk(bytesNotProcessedFromPreviousChunk.Span, isFinalBlock: true, out numBytesNotWrittenFromPreviousChunk);

```

Comment:
```suggestion
                    this.jsonWriter.Base64EncodeAndWriteChunk(bytesNotProcessedFromPreviousChunk.Span, isFinalBlock: true, out this.numBytesNotWrittenFromPreviousChunk);
```
Also see my comments elsewhere:
- https://github.com/OData/odata.net/pull/2880/files#r1551015622
- https://github.com/OData/odata.net/pull/2880/files#r1551020503